CharlieBackend.Business/Services/ScheduleServiceFolder/ScheduleService.cs

```diff
@@ -189,6 +190,32 @@ public async Task<Result<ScheduledEventDTO>> UpdateScheduledEventByID(long sched
             return Result<ScheduledEventDTO>.GetSuccess(_mapper.Map<ScheduledEventDTO>(item));
         }
 
+        public async Task<Result<ScheduledEventDTO>> AddSingleScheduledEvent(CreateScheduledEventDto createSingleScheduleRequest)
+        {
+            var singleScheduleRequest = new CreateScheduleDto()
+            {
+                Context = createSingleScheduleRequest.Context,
+                Range = createSingleScheduleRequest.Range
+            };
+
+            string error = await ValidateCreateScheduleEventRequestAsync(singleScheduleRequest);
+
+            if (error != null)
+            {
+                return Result<ScheduledEventDTO>.GetError(ErrorCode.ValidationError, error);
+            }
+
+            EventOccurrence result = null;
+
+            _unitOfWork.EventOccurrenceRepository.Add(result);
+
+            await _unitOfWork.CommitAsync();
+
+            await AddEventsAsync(result, singleScheduleRequest);
```

Comment:
You have to add one event. please use ScheduledEventRepository here instead of method